src/software/utils/main_imageProcessing.cpp

```diff
@@ -423,12 +427,24 @@ void processImage(image::Image<image::RGBAfColor>& image, const ProcessingParams
         }
     }
 
-    if (pParams.scaleFactor != 1.0f)
+    if ((pParams.scaleFactor != 1.0f) || (pParams.maxWidth != 0) || (pParams.maxHeight != 0))
```

Comment:
I would prefer to compute a scaleFactor based on pParams.scaleFactor, pParams.maxWidth, pParams.maxHeight.
And keep this if to check if this new scaleFactor is not 1.
As currently, if pParams.scaleFactor is one and maxWidth/Height are not 0 but a large value, it could introduce image duplications.